### PR TITLE
Fix USO update button

### PR DESCRIPTION
**Pull request by Mottie**
I'm not sure if this is a good fix...

It looks like USO fixed the md5Url `update.update` link (see #527), but now the JSON `originalMd5` doesn't always match the md5 in the md5-url:

* _GitHub-Dark_

  * [JSON `originalMd5`](https://userstyles.org/styles/chrome/37035.json) = `c07b885446fb71e6af18922aa58a5f3b`
  * [md5-url](https://update.userstyles.org/37035.md5) = `26c2370c297f99f8e5334c840e4b9e26`

* _GitHub-Fixed-Header_

  * [JSON `originalMd5`](https://userstyles.org/styles/chrome/124438.json) = `5a532095352f9e85cccf48b0b65268ac`
  * [md5-url](https://update.userstyles.org/124438.md5) = `5a532095352f9e85cccf48b0b65268ac`

### CODE REVIEW

**Comment by eight04** [comment]
This probably also affects `background/update.js` and `popup/search-results.js` since they also download style JSON from https://userstyles.org/styles/chrome/37035.json.

**Comment by Mottie** [comment]
This PR is ready for review/merging.

File: content/install-hook-userstyles.js [hunk 1]
```diff
@@ -79,6 +80,7 @@
     const md5Url = getMeta('stylish-md5-url');
     if (md5Url && installedStyle.md5Url && installedStyle.originalMd5) {
       getResource(md5Url).then(md5 => {
+        currentMd5 = md5;
```
Comment (eight04) [inline content/install-hook-userstyles.js]: Is it possible that `currentMd5` is `undefined` when `md5Url && installedStyle.md5Url && installedStyle.originalMd5` is not truthy?

File: background/update.js [hunk 2]
```diff
@@ -156,7 +156,8 @@
         }
         // USO can't handle POST requests for style json
         return download(style.updateUrl, {body: null})
-          .then(text => tryJSONparse(text));
+          // USO may not provide a correctly updated originalMd5 (#555)
+          .then(text => Object.assign(tryJSONparse(text), {originalMd5: md5}));
```
Comment (eight04) [inline background/update.js]: This would throw if `tryJSONparse(text)` returns `undefined`.

File: content/install-hook-userstyles.js [hunk 3]
```diff
@@ -169,7 +171,8 @@
           'https://github.com/openstyles/stylus/issues/195');
         return;
       }
-      return API.installStyle(Object.assign(json, addProps))
+      // Update originalMd5 since USO changed it (2018-11-11) to NOT match the current md5
+      return API.installStyle(Object.assign(json, addProps, {originalMd5: currentMd5}))
```
Comment (eight04) [inline content/install-hook-userstyles.js]: We may have to ensure `currentMd5` is set before calling `API.installStyle`.

File: popup/search-results.js [hunk 4]
```diff
@@ -578,6 +579,19 @@ window.addEventListener('showStyles:done', function _() {
           return result.style_settings;
         });
     }
+
+    function fetchMd5(result) {
+      return API.download({
+        url: UPDATE_URL.replace('%', result.id),
+        timeout: 60e3,
```
Comment (eight04) [inline popup/search-results.js]: Why passing these options? We only send the URL to download API in update.js:
https://github.com/openstyles/stylus/blob/3d3a2b802aaf00fc0d728822793dfd163b76f9b0/background/update.js#L150

File: content/install-hook-userstyles.js (hunk 1 above)
Comment (Mottie) [reply content/install-hook-userstyles.js]: `currentMd5` now set outside of the condition.

File: content/install-hook-userstyles.js (hunk 3 above)
Comment (Mottie) [reply content/install-hook-userstyles.js]: I'm not sure it still needs this since `currentMd5` is sure to be set now.

**Comment by eight04** [comment]
I've tested this on Chrome, and it seems that something doesn't work correctly. STR:

1. Visit https://userstyles.org/styles/37035/github-dark
2. Install style.
3. Open manager, click "Check all styles for updates".
4. Switch back to USO, press F5.
5. The page shows the install button again.
6. Click the install button, the style would be installed again without overwriting the old style.

**Comment by Mottie** [comment]
The initial problem was that the `currentMd5` wasn't being set if the style wasn't already installed... it is now set earlier.

The problem with updating the button was that the "styleAlreadyInstalledChrome" event was fired before USO applied the listener. I added a 300ms delay (maybe that wasn't enough?) too allow React to render.

**Comment by eight04** [comment]
Tested again and can no longer repro https://github.com/openstyles/stylus/pull/555#issuecomment-439452397

**Pull request merged**